src/tags/object/TimeSeries.js

```diff
@@ -305,6 +307,17 @@ const Model = types
       }
       return self._format(time);
     },
+
+    formatDuration(duration) {
+      if (!self._formatDuration) {
+        const { timedisplayformat: format, isDate } = self;
+
+        if (format === "date") self._formatDuration = duration => moment.utc(duration).format("HH:mm:ss.SSS");
+        else if (format) self._formatDuration = isDate ? d3.timeFormat(format) : d3.format(format);
+        else self._formatDuration = String;
+      }
+      return self._formatDuration(duration);
+    },
```

Comment:
I'm not sure what this function does, it appears to just hardcode a time format whenever the `timedisplayformat` is "date". 

Maybe instead of doing that, add an explicit `durationdisplayformat` and  default it to `%H:%M:%S.%L` so it's at least independently configurable and not tied to the timedisplayformat implicitly. 

If we were to do that, we could also skip the check for `format === "date"` and the moment dependency entirely, and just use `d3.utcFormat(format)`.

```suggestion
    formatDuration(duration) {
      if (!self._formatDuration) {
        const { durationdisplayformat: format } = self;

        if (format) self._formatDuration = d3.utcFormat(format);
        else self._formatDuration = String;
      }
      return self._formatDuration(duration);
    },
```

Reply:
that's the copy-paste from existing `formatTime` function couple of lines before, and yes, this function is weird %)
especially in this comparison with "date" which is not in the doc and not in the defaults, so no one will use it...

Reply:
I agree with the general concept to parameterize the format and get rid of the `format === "date"`, but don't we still need some type of distinction between non-date and date values (for ".1f" vs "%H:%M")? Then it is still somewhat coupled to timedisplayformat, which needs to be set to `date` (due to the functionality of `isDate`).

```suggestion
    formatDuration(duration) {
      if (!self._formatDuration) {
        const { durationdisplayformat: format, isDate } = self;

        if (format) self._formatDuration = isDate ? d3.utcFormat(format) : d3.format(format);
        else self._formatDuration = String;
      }
      return self._formatDuration(duration);
    },
```

Reply:
That makes sense for non time based ranges, yes. I think that suggestion should cover the cases required.